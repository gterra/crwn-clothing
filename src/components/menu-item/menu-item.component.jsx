import React from "react";
import { withRouter } from "react-router-dom";

import {
  BackgroundImageDiv,
  MenuItemDiv,
  ContentDiv,
  TitleH1,
  SubTitleSpan,
} from "./menu-item.styles";

const MenuItem = ({ item, history, match }) => (
  <MenuItemDiv
    size={item.size}
    onClick={() => history.push(`${match.url}${item.linkUrl}`)}
  >
    <BackgroundImageDiv imageUrl={item.imageUrl} />
    <ContentDiv>
      <TitleH1>{item.title.toUpperCase()}</TitleH1>
      <SubTitleSpan>SHOP NOW</SubTitleSpan>
    </ContentDiv>
  </MenuItemDiv>
);

export default withRouter(MenuItem);
