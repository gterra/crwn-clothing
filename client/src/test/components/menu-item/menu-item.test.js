import React from "react";
import { shallow } from "enzyme";

import { MenuItem } from "../../../components/menu-item/menu-item.component";

describe("MenuItem component", () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const linkUrl = "/hats";
  const size = "large";
  const imageUrl = "testimage";

  beforeEach(() => {
    mockMatch = {
      url: "/shop",
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      item: { linkUrl, size, title: "hats", imageUrl },
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it("should render MenuItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push with the right string when MenuItemDiv clicked", () => {
    wrapper.find("MenuItemDiv").simulate("click");

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it("should pass size to MenuItemDiv as the prop size", () => {
    expect(wrapper.find("MenuItemDiv").prop("size")).toBe(size);
  });

  it("should pass imageUrl to BackgroundImageDiv as the prop imageUrl", () => {
    expect(wrapper.find("BackgroundImageDiv").prop("imageUrl")).toBe(imageUrl);
  });
});
