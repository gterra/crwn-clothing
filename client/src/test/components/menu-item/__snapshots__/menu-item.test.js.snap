// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
<MenuItemDiv
  onClick={[Function]}
  size="large"
>
  <BackgroundImageDiv
    imageUrl="testimage"
  />
  <ContentDiv>
    <TitleH1>
      HATS
    </TitleH1>
    <SubTitleSpan>
      SHOP NOW
    </SubTitleSpan>
  </ContentDiv>
</MenuItemDiv>
`;
